src/i18n/i18nConfig.ts

```diff
@@ -1,13 +1,18 @@
 import i18n from "i18next";
 import LanguageDetector from "i18next-browser-languagedetector";
 import { initReactI18next } from "react-i18next";
+import api from "~/common/lib/api";
 
 import en_common from "./locales/en/common.json";
 import hi_common from "./locales/en/common.json";
 import en_components from "./locales/en/components.json";
 import en from "./locales/en/translation.json";
 import hi from "./locales/hi/translation.json";
 
+api.getSettings().then((setting) => {
```

Comment:
We're worried that this could lead to a race condition. Depending on how long it takes to get the settings this works or might fail and the LanguageDetector wins.  
Mayeb there's a way to combine this to make sure it's save to use?
@pavanjoshi914 what do you think?